homeassistant/components/esphome/lock.py

```diff
@@ -0,0 +1,83 @@
+"""Support for ESPHome locks."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioesphomeapi import LockCommand, LockEntityState, LockInfo, LockState
+
+from homeassistant.components.lock import SUPPORT_OPEN, LockEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_CODE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import EsphomeEntity, platform_async_setup_entry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up ESPHome switches based on a config entry."""
+    await platform_async_setup_entry(
+        hass,
+        entry,
+        async_add_entities,
+        component_key="lock",
+        info_type=LockInfo,
+        entity_type=EsphomeLock,
+        state_type=LockEntityState,
+    )
+
+
+class EsphomeLock(EsphomeEntity[LockInfo, LockEntityState], LockEntity):
+    """A lock implementation for ESPHome."""
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return self._static_info.assumed_state
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return SUPPORT_OPEN if self._static_info.supports_open else 0
+
+    @property
+    def code_format(self) -> str | None:
+        """Regex for code format or None if no code is required."""
+        if self._static_info.requires_code:
+            return self._static_info.code_format
+        return None
+
+    @property
```

Comment:
Should be
```suggestion
    @esphome_state_property
```

To make it check first if state is set